qualtran/bloqs/state_preparation/state_preparation_alias_sampling.py

```diff
@@ -96,66 +97,76 @@ class StatePreparationAliasSampling(PrepareOracle):
         converter=lambda v: (v,) if isinstance(v, Register) else tuple(v)
     )
     alt: Union[Shaped, NDArray[np.int_]]
     keep: Union[Shaped, NDArray[np.int_]]
     mu: SymbolicInt
-    sum_of_lcu_coeffs: SymbolicFloat
+    sum_of_coefficients: SymbolicFloat
 
     @classmethod
-    def from_lcu_probs(
-        cls, lcu_probabilities: Sequence[float], *, probability_epsilon: float = 1.0e-5
+    def from_probabilities(
+        cls, unnormalized_probabilities: Sequence[float], *, precision: float = 1.0e-5
     ) -> 'StatePreparationAliasSampling':
-        """Factory to construct the state preparation gate for a given set of LCU coefficients.
+        r"""Factory to construct the state preparation gate for a given set of LCU coefficients.
+
+        Given input `unnormalized_probabilities` $w_l$, with sum $\lambda = \sum_l w_l$, this prepares
+        a state s.t. $p_l = \tilde{w_l} / \lambda$ such that the input `precision` $\epsilon$ satisfies:
+
+            $$
+                |w_l - \tilde{w}_l| \le \epsilon
+            $$
+
+        That is, the value `precision` is the absolute error in approximating the input values.
 
         Args:
-            lcu_probabilities: The LCU coefficients.
-            probability_epsilon: The desired accuracy to represent each probability
+            unnormalized_probabilities: The LCU coefficients $w_l$.
+            precision: The desired accuracy $\epsilon$ to represent each input value
                 (which sets mu size and keep/alt integers).
-                See `qualtran.linalg.lcu_util.preprocess_lcu_coefficients_for_reversible_sampling`
+                See `qualtran.linalg.lcu_util.preprocess_probabilities_for_reversible_sampling`
                 for more information.
         """
-        alt, keep, mu = preprocess_lcu_coefficients_for_reversible_sampling(
-            lcu_coefficients=lcu_probabilities, epsilon=probability_epsilon
+        sum_of_coefficients = sum(abs(x) for x in unnormalized_probabilities)
+        alt, keep, mu = preprocess_probabilities_for_reversible_sampling(
+            unnormalized_probabilities=unnormalized_probabilities,
+            epsilon=precision / sum_of_coefficients,
         )
-        N = len(lcu_probabilities)
+        N = len(unnormalized_probabilities)
         return StatePreparationAliasSampling(
             selection_registers=Register('selection', BoundedQUInt((N - 1).bit_length(), N)),
             alt=np.array(alt),
             keep=np.array(keep),
             mu=mu,
-            sum_of_lcu_coeffs=sum(abs(x) for x in lcu_probabilities),
+            sum_of_coefficients=sum_of_coefficients,
         )
 
     @classmethod
     def from_n_coeff(
-        cls,
-        n_coeff: SymbolicInt,
-        sum_of_lcu_coeffs: SymbolicFloat,
-        *,
-        probability_epsilon: SymbolicFloat = 1.0e-5,
+        cls, n_coeff: SymbolicInt, sum_of_terms: SymbolicFloat, *, precision: SymbolicFloat = 1.0e-5
```

Comment:
We should keep the naming consistent with the class attribute and factory method. The attribute has `sum_of_coefficients` and this has `sum_of_terms`.

Reply:
fixed